monai/handlers/nvtx_handlers.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Wrapper around NVIDIA Tools Extension for profiling MONAI ignite workflow
+"""
+
+from typing import TYPE_CHECKING, Optional, Sequence, Tuple, Union
+
+from monai.config import IgniteInfo
+from monai.utils import min_version, optional_import
+
+_nvtx, _ = optional_import("torch._C._nvtx", descriptor="NVTX is not installed. Are you sure you have a CUDA build?")
+if TYPE_CHECKING:
+    from ignite.engine import Engine, Events
+else:
+    Engine, _ = optional_import("ignite.engine", IgniteInfo.OPT_IMPORT_VERSION, min_version, "Engine")
+    Events, _ = optional_import("ignite.engine", IgniteInfo.OPT_IMPORT_VERSION, min_version, "Events")
+
+
+__all__ = ["RangeHandler", "RangePushHandler", "RangePopHandler", "MarkHandler"]
+
+
+class RangeHandler:
+    """
+    Attach a NVTX range to a pair of Ignite events.
+    It pushes an NVTX range at the first event and pops it at the second event.
+    Stores zero-based depth of the range that is started.
+
+    Args:
+        event: a string or pair of Ignite events to attach a range to
+            If a single string is provided, it should  describe the base name of a pair of default Ignite events
+            with _STARTED and _COMPLETED postfix (like "EPOCH" for EPOCH_STARTED and EPOCH_COMPLETED).
+            The common accepted events are: BATCH, ITERATION, EPOCH, and ENGINE.
+            For the complete list of Events,
+            check https://pytorch.org/ignite/generated/ignite.engine.events.Events.html.
+
+        msg: ASCII message to associate with range.
+            If not provided, the name of first event will be assigned to the NVTX range.
+    """
+
+    def __init__(self, events: Union[str, Tuple[Events, Events]], msg: Optional[str] = None) -> None:
+        if msg is None:
+            if isinstance(events, str):
+                msg = events
+            else:
+                msg = events[0].strip("_")[0]
```

Comment:
I didn't get what's the input data of this case `events[0].strip("_")[0]`?
Thanks.

Reply:
changed the logic.